include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -232,38 +232,35 @@ struct __parallel_scan_submitter;
 template <typename _CustomName, typename... _PropagateScanName>
 struct __parallel_scan_submitter<_CustomName, __internal::__optional_kernel_name<_PropagateScanName...>>
 {
-    template <typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _InitType, typename _LocalScan,
-              typename _GroupScan, typename _GlobalScan>
+    template <typename _Range1, typename _Range2, typename _InitType, typename _LocalScan, typename _GroupScan,
+              typename _GlobalScan>
     __future<sycl::event, __result_and_scratch_storage<typename _InitType::__value_type>>
-    operator()(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _InitType __init, _LocalScan __local_scan,
+    operator()(sycl::queue& __q, _Range1&& __rng1, _Range2&& __rng2, _InitType __init, _LocalScan __local_scan,
```

Comment:
Why are we passing queue by left reference? not by `&&`?
`&&` would allow to pass directly r-value queue like `exec.__get_queue()`.
The left reference forces us to create a queue instance on the stack first....

Reply:
I think you are not correct: in this case we will have an error like `error : no matching function for call to...` and the message about candidate like `note: candidate function template not viable: expects an rvalue for 2nd argument`

Reply:
That's exactly `sycl::queue&&` makes a more common solution, because it works for both cases:
1) sumitter{}(q_local); //is passed as an lvalue
2) sumitter{}(exec.queue()); ////is passed as an rvalue

And if I have a case (2), the writing `sycl::queue&&` doesn't force me to create an another instance  of queue `q_local` on the stack and after pass it further to a function which accepts  `sycl::queue&`.

Reply:
https://godbolt.org/z/35sjPdWb1

Reply:
Sorry, I meant  such way
```
template<typename Queue>
void fnc(Queue &&)
```

Reply:
In our de-facto agreement in all internal API for the `sycl-backend` we works with `sycl::queue` and never describe it as through template type. I believe we should align with this style in this PR too.
@akukanov what do you think?

Reply:
> we works with `sycl::queue` and never describe it as through template type

I don't argue with that..
I would like just to have two ways will work:
```
sumitter{}(q_local);
sumitter{}(exec.queue());
```
In the proposed code the second way doesnt work. That is my concern.

Reply:
I don't agreed with this proposal: we should pay more difficult code in submitters for that.
Also, submitters will have the same problem like they had before this PR: you approach will bring ability to have different instantiations of submitters depends on `sycl::queue` value category. Previously we had this problem `ExecutionPolicy` template parameter.

Reply:
Actually, my main point - I have a concern that way "submitter{}(exec.queue());" doesn't work.
Before the change I can write both ways
`submitter{}(exe_local); or submitter{}(obj.exec());`

Regarding to `more difficult code` - 
1) it depends how loot at that.
2) There is  a C/CPP rule - to write a function signature such way that is suitable  and convenient for a caller. 

Other words, I don't insist on the code should be changed.
I am just saying about my concern - sumitter{}(exec.queue()); way doesn't work. And I would get a simple answer why we OK(if OK) with that. For example, an answer "sumitter{}(exec.queue()); way is not a value for oneDPL code" will suit me.

Reply:
Update: "different instantiations of submitters depends" also suits me